test/integration/concurrency_controls_test.rb

```diff
@@ -183,7 +183,46 @@ class ConcurrencyControlsTest < ActiveSupport::TestCase
     assert job.reload.ready?
   end
 
+  if ActiveRecord::Base.connection.adapter_name == "PostgreSQL"
+    test "insert_with_unique_by_has_same_database_results_as_create!_with_exception_handling" do
+      key = "key", limit = 1, expires_at = 1.minute.from_now
+
+      assert SolidQueue::Semaphore.count == 0
+
+      SolidQueue::Semaphore.insert({ key: key, value: limit - 1, expires_at: expires_at }, unique_by: :key)
+      assert SolidQueue::Semaphore.count == 1
+
+      SolidQueue::Semaphore.insert({ key: key, value: limit - 1, expires_at: expires_at }, unique_by: :key)
+      assert SolidQueue::Semaphore.count == 1
+
+      SolidQueue::Semaphore.delete_all
+      assert SolidQueue::Semaphore.count == 0
+
+      SolidQueue::Semaphore.create!(key: key, value: limit - 1, expires_at: expires_at)
+      assert SolidQueue::Semaphore.count == 1
+
+      assert_raises ActiveRecord::RecordNotUnique do
+        SolidQueue::Semaphore.create!(key: key, value: limit - 1, expires_at: expires_at)
+      end
+    end
+  end
```

Comment:
Hmm... actually, I'm not completely sure if perhaps we need this test at all, since it seems it's testing more something internal of Active Record, rather than something we have in Solid Queue 🤔

Reply:
While this really is more of a test of AR, its value is that we have a confirmation of outcomes with the two different ways to add records to the Semaphore table and we show the two mechanisms "side-by-side".  I realized that it was missing an assert to meet that goal.  See what you think with the extra line.  

In the end, if you still feel strongly about cleaning this test up a little, I'll remove it.

Reply:
Sounds good! I don't feel strongly about this one 😊